pyveda/vedaset/stream/vedastream.py

```diff
@@ -251,12 +255,14 @@ def from_vc(cls, vc, **kwargs):
                    vc.imshape, **kwargs)
 
     def __enter__(self):
+
         self._start_consumer()
+        while len(self.train) + len(self.test) + len(self.validate) < 1:
```

Comment:
We just need @rjpolackwich eyes on this when he's back

Reply:
yeah, it's a bit of hack for now, __getitem__ should be doing the waiting

Reply:
`__getitem__` now does the waiting, but the question is does the buffer know to keep going to get to the index. Or should this just do an API call?